features/post_url_tag.feature

```diff
@@ -0,0 +1,144 @@
+Feature: PostUrl Tag
```

Comment:
Since we can also use the `{% link ... %}` tag to link to posts (which is not being touched upon in this feature-set), I felt that this heading added the *encapsulation expected* from feature tests.

We already have a dedicated feature-set for the behavior surrounding our `{% link ... %}` here:
https://github.com/jekyll/jekyll/blob/9e42fad62fed89c1aa3662cac9a01f4488695d58/features/link_tag.feature#L1-L4

Reply:
we should merge those two then, Matt is right here.

Reply:
In that case the merging can be done via a separate PR to keep the diff here focused.

Reply:
There is a difference between `post_url` and `link` that I think is important to call out here, and that difference is that `link` requires a path relative to the site root to link to. The reason I titled the feature `Linking to posts` was because we're being very explicit that this is only about posts, and not anything else, and that this is a specific way to handle posts for your site.

If we want to do any merging, then I say we wait for later.

Reply:
This is tech debt, if we were to refactor links in Jekyll, we wouldn't have both. 🤷‍♂️

Reply:
Refactoring links in Jekyll would be another breaking change would it not?